src/runner/runner.ts

```diff
@@ -112,15 +112,69 @@ export interface RunnerProps {
   network?: NetworkProps;
 
   manager?: {
-    machineImage?: IMachineImage; // An Amazon Machine Image ID for the Manager EC2 instance.
-    instanceType?: InstanceType; // Instance type for manager EC2 instance. It's a combination of a class and size.
-    keyPairName?: string; // A set of security credentials that you use to prove your identity when connecting to an Amazon EC2 instance. You won't be able to ssh into an instance without the Key Pair.
+    /**
+     * An Amazon Machine Image ID for the Manager EC2 instance.
+     */
+    machineImage?: IMachineImage;
+    /**
+     * Instance type for manager EC2 instance. It's a combination of a class and size.
+     * @default InstanceType.of(InstanceClass.T3, InstanceSize.NANO)
+     */
+    instanceType?: InstanceType;
+    /**
+     * A set of security credentials that you use to prove your identity when connecting to an Amazon EC2 instance. You won't be able to ssh into an instance without the Key Pair.
+     */
+    keyPairName?: string;
   };
-
   runners?: {
-    instanceType?: InstanceType; // Instance type for runner EC2 instances. It's a combination of a class and size.
-    machineImage?: IMachineImage; // An Amazon Machine Image ID for the Runners EC2 instances.
+    /**
+     * Instance type for runner EC2 instances. It's a combination of a class and size.
+     * @default InstanceType.of(InstanceClass.T3, InstanceSize.MICRO)
+     */
+    instanceType?: InstanceType; //
+    /**
+     * An Amazon Machine Image ID for the Runners EC2 instances.
+     */
+    machineImage?: IMachineImage; //
+    /**
+     * This defines the Docker Container parameters.
+     * https://docs.gitlab.com/runner/configuration/advanced-configuration.html#the-runnersdocker-section
+     *
+     * @example
+     * ```
+     * {
+     *   tls_verify: false,
+     *   image: "docker:19.03.5",
+     *   privileged: true,
+     *   cap_add: ["CAP_SYS_ADMIN"],
+     *   wait_for_services_timeout: 300,
+     *   disable_cache: false,
+     *   volumes: ["/certs/client", "/cache"],
+     *   shm_size: 0,
+     * }
+     * ```
+     */
     docker?: Partial<DockerConfiguration>;
+    /**
+     * The following parameters define the Docker Machine-based autoscaling feature.
+     * https://docs.gitlab.com/runner/configuration/advanced-configuration.html#the-runnersmachine-section
+     *
+     * @example
+     * ```
+     * {
+     *   IdleCount: 0,
+     *   IdleTime: 300,
+     *   MaxBuilds: 20,
+     *   MachineDriver: "amazonec2",
+     *   MachineName: "gitlab-runner-%s",
+     *   autoscaling: {
```

Comment:
It's easier to Omit autoscaling rules here and paste as an optional array. In case the optional array is empty, the default get set.